pool.go

```diff
@@ -69,10 +79,11 @@ type pool struct {
 
 func newPool(c *Config) *pool {
 	p := pool{
-		config:  c,
-		started: make(chan struct{}),
-		refresh: make(chan struct{}, 1),
-		done:    make(chan struct{}, 1),
+		config:        c,
+		started:       make(chan struct{}),
+		refresh:       make(chan struct{}, 1),
+		done:          make(chan struct{}, 1),
+		mirrorSamples: make(chan poolRequest, 10),
```

Comment:
why this magic number ?

Reply:
it gives some slack for parallelism / how much of a backlog of requests are we letting build up - should be more than 1 probably, but shouldn't be huge